src/common/login/PostLoginActions.ts

```diff
@@ -115,8 +116,51 @@ export class PostLoginActions implements PostLoginAction {
 		this.showSetupWizardIfNeeded()
 	}
 
+	// Runs the user approval check after the user has been updated or after a timeout
+	private checkApprovalAfterSync(): Promise<void> {
+		return new Promise<void>((resolve) => {
+			// Keep track of whether the listener was removed due to an entity update
+			let wasListenerRun = false
+
+			// Add an event listener to run the check after any customer entity update
+			const listener = async (updates: ReadonlyArray<EntityUpdateData>) => {
+				// Get whether the entity update contains the customer
+				let isCustomerUpdate: boolean = false
+				const customer = this.logins.getUserController().user.customer
+				for (const update of updates) {
+					if (
+						update.operation === OperationType.UPDATE &&
```

Comment:
We could perhaps `updates.some(u => isUpdateFor(u, customer))`

Reply:
Ah, I did not notice that function. It looks cleaner that way for sure. `isUpdateFor()` requires the customer object rather than just the ID so it must be loaded. It might add a small delay but likely will not effect much.

Reply:
ah I actually didn't notice that we don't have a customer yet, I'm sorry. perhaps then we should use `isUpdateForTypeRef()` and a separate ID check.